File: day19/day_nineteen.cpp
# include <iostream>
# include <unordered_set>

using namespace std;

string arrayContainsPairSum(int arr[],int n,int x)
{
	unordered_set<int> us;
	for(int i=0;i<n;i++)
	{
		if(us.find(x-arr[i]) == us.end())
		{
			us.insert(arr[i]);
		}
		else
		{
			return "arrayContainsPairSum";
		}
	}
	return "Doesn't contains PairSum";
}

int main()
{
	int arr[] = {1,6,7,8,20,100};
	int x;
	x = 12;
	cout << "x = " << x << "," << arrayContainsPairSum(arr,sizeof(arr)/sizeof(arr[0]),x) << endl;
	x = 15;
	cout << "x = " << x << "," << arrayContainsPairSum(arr,sizeof(arr)/sizeof(arr[0]),x) << endl;

	return 0;
}

File: day19/DayNineteen.java
import java.util.*;

public class DayNineteen
{
	public static void main(String args[])
	{
		DayNineteen subArrSum = new DayNineteen();
		int arr[] = {1,6,7,8,20,100};
		int x;
		x = 12;
		System.out.println("x = " + x + "," + subArrSum.arrayContainsPairSum(arr,arr.length,x));
		x = 15;
		System.out.println("x = " + x + "," + subArrSum.arrayContainsPairSum(arr,arr.length,x));
	}

	private String arrayContainsPairSum(int arr[],int n,int x)
	{
		Set<Integer> hset = new HashSet<Integer>();
		for(int i=0;i<n;i++)
		{
			if(hset.contains(x-arr[i]) == false)
			{
				hset.add(arr[i]);
			}
			else
			{
				return "arrayContainsPairSum";
			}
		}
		return "Doesn't contains PairSum";
	}

}